hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HoodieCatalog.java

```diff
@@ -480,8 +481,8 @@ public void dropPartition(ObjectPath tablePath, CatalogPartitionSpec catalogPart
     }
 
     try (HoodieFlinkWriteClient<?> writeClient = HoodieCatalogUtil.createWriteClient(options, tablePathStr, tablePath, hadoopConf)) {
-      writeClient.deletePartitions(Collections.singletonList(partitionPathStr),
-              writeClient.createNewInstantTime())
+      String instantTime = writeClient.startCommit(HoodieTimeline.REPLACE_COMMIT_ACTION);
```

Comment:
nts: this line will do the state change safely i.e request the write first

Reply:
This is not right, the requested instant was created with empty `HoodieRequestedReplaceMetadata`.